src/broadcast/broadcast_manager.ts

```diff
@@ -305,10 +356,33 @@ export class BroadcastManager extends EventEmitter {
       });
 
       getBroadcasts().catch(console.warn);
+      const postSocketConnectingSubStepRetry = () => {
```

Comment:
this function is giving me pause atm. could you add some comments and explain why this needs to run constantly with `setTimeout` and instead of only if `getBroadcasts` fails?

Reply:
`getBroadcasts` and `startBroadcast` only fail if something goes wrong client-side, that is, if we fail to send the message. We expect the server to respond to both of these messages but because WebSockets simply deals with 'messages' or 'events' and doesn't use a request/response paradigm, there no notion of the server failing to respond to a message. Setting a simple exponential backoff retry covers all possible cases:

1. failure to send
2. server sends an unexpected/invalid/error response
3. server sends nothing